enhancements/monitoring/alert-overrides.md

```diff
@@ -0,0 +1,322 @@
+    rules:
+    - alert: CustomAlert
+      expr: "vector(1)"
+      for: 15m
+      labels:
+        custom: "true"
+```
+
+A new controller in the cluster-monitoring-operator will generate
+`PrometheusRule` resources based on the `AlertingRule` instances in the
+`openshift-monitoring` namespace.  These will be consumed by the platform
+prometheus-operator instance like any other `PrometheusRule`.  All user-defined
+alerts will be labeled identifying them as such -- i.e. with a static
+`openshift_io_user_alert=true` label.
+
+The new `AlertRelabelConfig` type will allow relabeling operations on alerts.
+All `AlertRelabelConfig` instances will be collected in order to build a single
+secret with additional alert relabel configs used by the prometheus-operator.
+Example of changing an alert severity from `none` to `critical`:
+
+```yaml
+---
+apiVersion: monitoring.openshift.io/v1alpha1
+kind: AlertRelabelConfig
+metadata:
+  labels:
+    app.kubernetes.io/instance: k8s
+  name: watchdog
+  namespace: openshift-monitoring
+spec:
+  configs:
+  - sourceLabels: [alertname, severity]
+    regex: "Watchdog;none"
+    targetLabel: severity
+    replacement: critical
+    action: replace
+```
+
+A new controller in the cluster-monitoring-operator will combine all
+`AlertRelabelConfig` instances in the `openshift-monitoring` namespace into a
+single secret, concatenating each resource in lexigraphical order by name.  This
+secret will be referenced in the `additionalAlertRelabelConfigs` parameter for
+the platform Prometheus instance.
+
+The `AlertRelabelConfig` controller may optionally disable certain actions like
+`drop` to prevent users disabling critical platform alerts.  Silences already
+exist as a method of ignoring unwanted alerts.
+
+Note that alert relabel operations only affect the alerts just before they are
+sent to Alertmanager -- the changes are not reflected in the Prometheus alerts
+API.  That means that currently dropped alerts will still be shown in the
+OpenShift console, though they won't be forwarded to Alertmanager.  It also
+means modifications to alerts, for example changed severities, are currently not
+reflected in the console.  This may necessitate some UI/UX work in the console.
+
+### User Stories
+
+- As an OpenShift user, I want to change the severity of an existing alert,
+  choosing between, e.g. from `Warning` to `Critical`.
+- As an OpenShift user, I want to add labels to alerts to aide in routing and
+  triage.
+- As an OpenShift user, I want a supported way to add custom alerts using
+  metrics from components in the `openshift-*` namespaces.
+- As an OpenShift developer, I want the alerting rules shipped by my component
+  to be evaluated and forwarded to telemetry even when overridden by a user.
+
+### API Extensions
+
+We will be adding two new CRDs for alerting rules and relabeling:
+
+- `alertingrule.monitoring.openshift.io/v1alpha1`
+- `alertrelabelconfig.monitoring.openshift.io/v1alpha1`
```

Comment:
Notice you've got these as v1alpha1 APIs at the moment, do you have a graduation plan for when these will become beta and eventually GA?

Typically, OpenShift APIs are added directly as a `v1` API and do not go through the API alpha and beta cycles

Reply:
I chose that more or less arbitrarily. The plan is for this to hopefully be tech-preview in 4.11 and go from there. Should I move them to `v1` even for tech-preview?